### ISSUE TITLE
[FEATURE REQUEST] Leveler

### ISSUE THREAD

**Issue by hlubach** [Akylas/OSS-DocumentScanner#122]
To make the best possible picture of a document 
You have to hold the phone as right as possible again the doc.
Therefore I suggest to add a leverer to this piece of software

**Comment by farfromrefug** [comment]
@hlubach do you hâve some example app doing that ?

**Comment by hlubach** [comment]
> @hlubach do you hâve some example app doing that ?

I know that camscanner has this function in it altough you must enable it in the setting.
In de top you then get one big circle and a smaller one with a mark that must be in the smaallest

**Comment by farfromrefug** [comment]
@hlubach we are talking about something like this? 
You want a tool help you ensure your phone is as "level" as the ground as possible?
![image](https://github.com/Akylas/OSS-DocumentScanner/assets/655344/d59bc441-549f-4376-bf3b-79a28c1dc5f4)

**Comment by HiFiJ** [comment]
Hi,
I imagine this is about being able to correct the document angle a couple of degrees after taking a picture. I also find it difficult to "scan"/take a photo so that the document comes out horizontally in the end. Mostly this is about being a couple of degrees off (say 1-5 degrees). In my workflow on my desktop PC I added a step to fine-tune the scanned image before OCR and PDF processing. This helps tesseract and produces more professional looking scans (PDFs).

Thanks very much for the cool app, it's much needed to have a good quality open source option!

Joost

**Comment by farfromrefug** [comment]
@HiFiJ Yes i actually felt the same thing using it this afternoon. 
So a leveler would solve your case too?
I think I will add one. Just need to find the right UX

**Comment by HiFiJ** [comment]
Hi,
sorry for the late reply!
Just to clarify what I mean (and use in my desktop scanning process with the Linux/KDE application "showfoto", a lightweight image editor):
To have a way to correct the rotation/angle slightly after taking the photo/scanning the image, i.e. 1) Take the photo 2) Get the possibility to rotate the image, for example with a dial wheel. Here, sub-degree steps would probably be necessary to get text rows adequalty horizontally oriented in the final document (PDF scan).

In contrast, I do not think that a "bubble level" to help taking a staight photo will help a lot (like in the post 2 above this one), as the photos will come out skewed still due to tiny hand movements.

Thanks for considering this feature!

**Comment by HiFiJ** [comment]
Here is an image roughly giving an idea of what I mean - with the dial below the image one could rotate the image left and right to get photographed text rows horizontally (disregard the fish-eye correction shown here - just an example for the UI).
 I imagine this would benefit OCR results as well as helping creating professional looking documents.

![image](https://github.com/Akylas/OSS-DocumentScanner/assets/8113097/a9343156-655e-418c-9979-35fd5d4b2617)

**Comment by farfromrefug** [comment]
@HiFiJ when you say rotation then we only talk about one axis rotation?
The leveler as you would show it in a leveler app would show 2 axis. 
Plus the 2 axis you would get are different from the one you would change with you UI (we could call him x axis).
To see If I understand your issue correctly, would a rotate image, like the one we already have with  90 degrees steps, but with a very fine grain, would br what you want?

**Comment by hlubach** [comment]
> @hlubach we are talking about something like this? You want a tool help you ensure your phone is as "level" as the ground as possible? ![image](https://private-user-images.githubusercontent.com/655344/303839017-d59bc441-549f-4376-bf3b-79a28c1dc5f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3MDI1NTgsIm5iZiI6MTcxOTcwMjI1OCwicGF0aCI6Ii82NTUzNDQvMzAzODM5MDE3LWQ1OWJjNDQxLTU0OWYtNDM3Ni1iZjNiLTc5YTI4YzFkYzVmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyOVQyMzA0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNGFiZTliZGFlMjdhYjhiY2NiOTc4MmQ5NjIzNjg1ZWU2OTBlOWQxYmI3ZDA2MDcwYjc3MzBlMDY2Mjg4NzY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.85biCPnmS4_EQRKJDfxOUksQRN7RL1kfqiQwN3I7Nhc)

it seem to be called bubble leveler

**Comment by farfromrefug** [comment]
@hlubach yes i am talking about bubble leveler like you would fine in https://github.com/kylecorry31/Trail-Sense
What @HiFiJ is showing in his screenshot seems more like a rotate image tool with fine grain (which i dont have right now).

**Comment by HiFiJ** [comment]
> @hlubach yes i am talking about bubble leveler like you would fine in https://github.com/kylecorry31/Trail-Sense What @HiFiJ is showing in his screenshot seems more like a rotate image tool with fine grain (which i dont have right now).

Yes, that would be what I mean. I believe it will always stay (very) difficult to take completely  straight shots from hand (which is also very problematic with my scanner - scans are always a couple of degrees off), therefore I propose to implement a quick way to correct this immediately after taking the picture.

**Comment by farfromrefug** [comment]
@HiFiJ Ok i get it now. Will have to see how I can implement that UX wise so that it works with the cropping tool if you want it to happen on scan "confirmation"
